apps/settings/src/components/AppStoreSidebar/AppDeployOptionsModal.vue

```diff
@@ -0,0 +1,322 @@
+<!--
+  - SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<NcModal v-if="show"
+		label-id="form-name"
+		@close="() => $emit('update:show', false)">
+		<div class="modal__content">
+			<h2 id="form-name">
+				{{ t('settings', 'Advanced deploy options') }}
+			</h2>
+			<p class="description" style="text-align: center;">
+				{{ configuredDeployOptions === null ? t('settings', 'Edit ExApp deploy options before installation') : t('settings', 'Configured ExApp deploy options. Can be set only during installation') }}.
+				<a href="https://docs.nextcloud.com/server/latest/admin_manual/exapps_management/AdvancedDeployOptions.html">Learn more</a>
+			</p>
+
+			<h3 v-if="environmentVariables.length > 0 || (configuredDeployOptions !== null && configuredDeployOptions.environment_variables.length > 0)">
+				{{ t('settings', 'Environment variables') }}
+			</h3>
+			<template v-if="configuredDeployOptions === null">
+				<div v-for="envVar in environmentVariables"
+					:key="envVar.envName"
+					class="deploy-option">
+					<NcTextField :label="envVar.displayName" :value.sync="deployOptions.environment_variables[envVar.envName]" />
+					<p class="description">
+						{{ envVar.description }}
+					</p>
+				</div>
+			</template>
+			<template v-else-if="Object.keys(configuredDeployOptions).length > 0">
+				<p class="description">
+					{{ t('settings', 'ExApp container environment variables') }}
+				</p>
+				<ul class="envs">
+					<li v-for="envVar in Object.keys(configuredDeployOptions.environment_variables)" :key="envVar">
+						<NcTextField :label="configuredDeployOptions.environment_variables[envVar].displayName ?? envVar"
+							:value="configuredDeployOptions.environment_variables[envVar].value"
+							readonly />
+						<p class="description">
+							{{ configuredDeployOptions.environment_variables[envVar].description }}
+						</p>
+					</li>
+				</ul>
+			</template>
```

Comment:
For accessibility and semantic correctness this makes more sense in a `fieldset`:

```suggestion
			<fieldset v-else-if="Object.keys(configuredDeployOptions).length > 0">
				<legend class="description">
					{{ t('settings', 'ExApp container environment variables') }}
				</legend>
				<NcTextField v-for="(value, key) in configuredDeployOptions.environment_variables"
					:key="key"
					:label="value.displayName ?? key"
					:helper-text="value.description"
					:value="value.value"
					readonly />
			</template>
```